File: leetcode/AIcoding/2.Gemini-2.0/cpp_solutions/3081-replace-question-marks-in-string-to-minimize-its-value.cpp
#include <string>

using namespace std;

class Solution {
public:
    string modifyString(string s) {
        int n = s.size();
        for (int i = 0; i < n; ++i) {
            if (s[i] == '?') {
                for (char c = 'a'; c <= 'c'; ++c) {
                    bool valid = true;
                    if (i > 0 && s[i - 1] == c) {
                        valid = false;
                    }
                    if (i < n - 1 && s[i + 1] != '?' && s[i + 1] == c) {
                        valid = false;
                    }
                    if (valid) {
                        s[i] = c;
                        break;
                    }
                }
            }
        }
        return s;
    }
};

File: leetcode/AIcoding/2.Gemini-2.0/rust_solutions/3081-replace-question-marks-in-string-to-minimize-its-value.rs
impl Solution {
    pub fn modify_string(s: String) -> String {
        let mut chars: Vec<char> = s.chars().collect();
        let n = chars.len();

        for i in 0..n {
            if chars[i] == '?' {
                for c in ['a', 'b', 'c'].iter() {
                    let mut valid = true;
                    if i > 0 && chars[i - 1] == *c {
                        valid = false;
                    }
                    if i < n - 1 && chars[i + 1] != '?' && chars[i + 1] == *c {
                        valid = false;
                    }

                    if valid {
                        chars[i] = *c;
                        break;
                    }
                }
            }
        }

        chars.into_iter().collect()
    }
}